src/openrct2/ride/Vehicle.cpp

```diff
@@ -6638,163 +6579,191 @@ void Vehicle::UpdateAnimationAnimalFlying()
-            targetFrame = (animationState >> 18) & 3;
-            if (animation_frame != targetFrame)
-            {
-                animation_frame = targetFrame;
-                Invalidate();
+                vehicle.animation_frame = targetSeatRotation;
+                vehicle.Invalidate();
             }
-            break;
-        case CAR_ENTRY_ANIMATION_MONORAIL_CYCLES: // Loc6D64B6
-            if (num_peeps != 0)
-            {
-                animationState += _vehicleVelocityF64E08;
-                eax = ((animationState >> 13) & 0xFF) << 2;
-                targetFrame = (eax >> 8) & 0xFF;
-                if (animation_frame != targetFrame)
-                {
-                    animation_frame = targetFrame;
-                    Invalidate();
-                }
-            }
-            break;
-        case CAR_ENTRY_ANIMATION_MULTI_DIM_COASTER: // Loc6D65E1
-            if (seat_rotation != target_seat_rotation)
-            {
-                if (animationState <= 0xCCCC)
-                {
-                    animationState += 0x3333;
-                }
-                else
-                {
-                    animationState = 0;
-
-                    if (seat_rotation >= target_seat_rotation)
-                        seat_rotation--;
+        }
+    }
+}
 
-                    else
-                        seat_rotation++;
+/**
+ * Animal Flying animates only on chainlift and in an unusual way. Made by Spacek531
+ */
+static void AnimateAnimalFlying(Vehicle& vehicle, const CarEntry& carEntry)
+{
+    vehicle.UpdateAnimationAnimalFlying();
+    // makes animation play faster with vehicle speed
+    uint8_t targetFrame = abs(_vehicleVelocityF64E08) >> carEntry.AnimationSpeed;
+    vehicle.animationState = std::max(vehicle.animationState - targetFrame, 0u);
+}
 
-                    animation_frame = (seat_rotation - 4) & 7;
-                    Invalidate();
-                }
-            }
-            break;
-        case CAR_ENTRY_ANIMATION_ANIMAL_FLYING:
-            UpdateAnimationAnimalFlying();
-            // makes animation play faster with vehicle speed
-            targetFrame = abs(_vehicleVelocityF64E08) >> 24;
-            animationState = std::max(animationState - targetFrame, 0u);
-            break;
+/**
+ *
+ *  rct2: 0x006D63D4
+ */
+void Vehicle::UpdateAdditionalAnimation()
+{
+    auto carEntry = Entry();
+    if (carEntry == nullptr)
+    {
+        return;
     }
+    if (carEntry->AnimationFrames == 0 || carEntry->animation >= CarEntryAnimation::Count)
+        return;
+    using AnimateFunction = void (*)(Vehicle & vehicle, const CarEntry& carEntry);
+    constexpr static AnimateFunction animationFunctions[EnumValue(CarEntryAnimation::Count)]{
+        AnimateNone,          AnimateSimpleVehicle,  AnimateSteamLocomotive,  AnimateSwanBoat,
+        AnimateMonorailCycle, AnimateMultiDimension, AnimateObservationTower, AnimateAnimalFlying
+    };
```

Comment:
I personally think this should be outside the function, also if you add a comma to the end clang-format will have this like a table, in theory.

Reply:
I can do that. Trailing comma does not change formatting though.

Reply:
Too bad, at least outside the function it doesn't really get in the way of reading the code, thanks.